lib/ansible/modules/cloud/azure/azure_rm_virtualmachine_facts.py

```diff
@@ -0,0 +1,393 @@
+                                "path": "/home/ubuntu/.ssh/authorized_keys"
+                            }
+                        ]
+                    }
+                },
+                "secrets": []
+            },
+            "provisioningState": "Succeeded",
+            "storageProfile": {
+                "dataDisks": [],
+                "imageReference": {
+                    "offer": "UbuntuServer",
+                    "publisher": "Canonical",
+                    "sku": "16.04-LTS",
+                    "version": "16.04.201801220"
+                },
+                "osDisk": {
+                    "caching": "ReadOnly",
+                    "createOption": "FromImage",
+                    "diskSizeGB": 30,
+                    "name": "vm.vhd",
+                    "osType": "Linux",
+                    "vhd": {
+                        "uri": "https://ubuntu1842.blob.core.windows.net/vhds/vm.vhd"
+                    }
+                }
+            },
+            "vmId": "77f0006c-0874-42ca-9ec9-e920b36263cf"
+        },
+        "resources": [
+            {
+                "id": "/subscriptions/XXXXX/resourceGroups/Testing/providers/Microsoft.Compute/virtualMachines/vm/extensions/OmsAgentForLinux",
+                "location": "brazilsouth",
+                "name": "OmsAgentForLinux",
+                "properties": {
+                    "autoUpgradeMinorVersion": true,
+                    "provisioningState": "Succeeded",
+                    "publisher": "Microsoft.EnterpriseCloud.Monitoring",
+                    "settings": {
+                        "azureResourceId": "/subscriptions/XXXXX/resourcegroups/Testing/providers/microsoft.compute/virtualmachines/vm",
+                        "stopOnMultipleConnections": true,
+                        "workspaceId": "43ac2095-42c7-4488-924f-bc4cbca63f8d"
+                    },
+                    "type": "OmsAgentForLinux",
+                    "typeHandlerVersion": "1.0"
+                },
+                "type": "Microsoft.Compute/virtualMachines/extensions"
+            }
+        ],
+        "tags": {
+            "cluster_name": "cluster"
+        },
+        "type": "Microsoft.Compute/virtualMachines"
+    }]
+'''
+
+try:
+    from msrestazure.azure_exceptions import CloudError
+except:
+    # This is handled in azure_rm_common
+    pass
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase, azure_id_to_dict
+
+
+AZURE_OBJECT_CLASS = 'VirtualMachine'
+
+AZURE_ENUM_MODULES = ['azure.mgmt.compute.models']
+
+
+class AzureRMVirtualMachineFacts(AzureRMModuleBase):
+
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            resource_group=dict(type='str', required=True),
+            tags=dict(type='list'),
+            auth_source=dict(type='str', choices=['auto', 'cli', 'env', 'credential_file', 'msi'], default='auto'),
+            cloud_environment=dict(type='str', default='AzureCloud')
```

Comment:
`auth_source` and `cloud_environment` are already in base class, remove them here.

Reply:
It is fine to add the two options here right now, but the correct way is to fix the base class. I will propose another pr to fix that.